src/main/java/com/google/cloud/tools/maven/CloudSdkAppEngineFactory.java

```diff
@@ -241,6 +252,25 @@ public void onOutputLine(String line) {
     }
   }
 
+  /** A listener that redirects process output to a file. */
+  protected static class FileOutputLineListener implements ProcessOutputLineListener {
```

Comment:
Is the protected visibility necessary?

Reply:
I was mimicking the `DefaultProcessOutputLineListener` definition, but yeah it looks like both can be package private.